templates/cc/repeated.go

```diff
@@ -24,15 +24,23 @@ const repTpl = `
 	{{ end }}
 
 	{{ if $r.GetUnique }}
-	std::set<std::reference_wrapper<const {{ $typ }}>> {{ lookup $f "Unique" }};
+	// Implement comparison for wrapped reference types
+	struct cmp {
```

Comment:
Shouldn't we just use `std::unordered_set` below?

Reply:
I switched to `std::unordered_set` for efficiency, but it still has the same problems that `std::hash` and `std::equal_to` don't just do the right thing.